app/assets/javascripts/pages/referenceWidget.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * Widget to select an existing reference
+ */
+(function (root, factory) {
+  if (typeof module === 'object' && module.exports) {
+    module.exports = factory(require('jQuery'), require('../common/utility'));
+  } else {
+    root.ExistingReferenceWidget = factory(root.jQuery, root.utility);
+  }
+}(this, function ($, utility) {
+  // CONSTANTS
+  var REFERENCES_PER_PAGE = 75;
+  var REQUEST_URL = '/references/recent';
+  var TYPEAHEAD_OPTIONS = {
+    minLength: 2,
+    highlight: true
+  };
+    
+  /**
+   * Reference Widget
+   * Use:  new ReferenceWidget([entity_ids], { options })
+   * @param {Array|Integer|String} entityIds
+   * @param {Object} userOptions
+   */
+  function ExistingReferenceWidget(entityIds, userOptions) {
+    this.options = mergeOptions(ExistingReferenceWidget.DEFAULT_OPTIONS, userOptions);
+    this.entityIds = [].concat(entityIds).map(function(n) { return Number(n); });
```

Comment:
they could have just as easily been strings here, since ultimately they get serialized as json and sent to rails which does it's own thing with entity ids. but since they _are_ numbers in the database, i usually keep them as numbers in the client unless there's a reason (such as the situation you are talking about)